rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -472,11 +504,9 @@ private NodeSet selectRecursively(int number, TreeSet<Host> freeHosts) {
             // complexity is log(n)
             InetAddress closestHost = removeClosest(number, freeHosts);
 
-            List<Node> nodes = nodesOnHost.get(closestHost);
+            LinkedHashSet<Node> nodes = nodesOnHost.get(closestHost);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "nodes" object should be an interface such as "Set" rather than the implementation "LinkedHashSet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)